### ISSUE TITLE
Overwrite defaultSeverity from the command line

### ISSUE THREAD

**Issue by fabienchn** [palantir/tslint#4557]
I have a `defaultSeverity: "warning"`in my tslint config because I don't want the linter to be a blocker during development.
However, before deployment, when my pipeline is running, I want the defaultSeverity to be "error". I want my pipeline to fail if there is any linting issue.

So I thought there would be a way to overwrite the `tslint.json` config file by running something like:
`yarn tslint -c tslint.json defaultSeverity=error`

But didn't find this anywhere in the docs.

Any idea if this already exists ? 
If not, what do you think of adding this feature ?

**Comment by JoshuaKGoldberg** [comment]
Seems like a great idea! To match up with the existing CLI flags in `tslintCli.ts`, how about?:

```shell
yarn tslint -c tslint.json --defaultSeverity "error"
```

**Comment by fabienchn** [comment]
Anyone who could do it ? I tried it, but didn't manage to really understand how the internal stuff works...

**Comment by fabienchn** [comment]
I found a solution to this use case. 

`./tslint-strict.json`:
```
{
  "extends": "./tslint.json",
  "defaultSeverity": "error"
}
```

But having a new file just for a tiny line of config feels wrong...

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
